src/app/SafeAttributePersistenceProvider.h

```diff
@@ -141,7 +154,7 @@ class SafeAttributePersistenceProvider
      * @param [in] aPath the attribute path for the data being written.
      * @param [in] aValue the data to write. The value will be stored as-is.
      */
-    virtual CHIP_ERROR SafeWriteValue(const ConcreteAttributePath & aPath, const ByteSpan & aValue) = 0;
+    virtual CHIP_ERROR SafeWriteValue(const ConcreteAttributePath & aPath, const ByteSpan & aValue);
```

Comment:
No, the whole point of SafeAttributePersistenceProvider is that it's an interface, which can be implemented however is convenient to the application, no?  In particular, it might not be implemented on top of PersistentStorageDelegate at all!  

I'm stopping here, since this seems like a pretty fundamental issue.  So just to be clear:

1. This is meant to be a pure interface.
2. Some specific implementations of this interface might exist, and the SDK should provide a default one to make things easy for people who want to use that default.
3. The implementation to actually use should be injected.  I know that's not what we do now, and Server just hardcodes a `DefaultAttributePersistenceProvider`, but that's a TODO/bug to fix, not something to reify in our API surface.

Reply:
Following up offline to figure out an approach here. This is what we have today (has a very organic-grown feel to it):

![image](https://github.com/user-attachments/assets/5943b1f4-3814-4e81-b6d2-7e13428228ba)